crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +140,133 @@ impl Persistence {
             .await
             .map_err(Error::DbError)
     }
+
+    /// Get auction data.
+    pub async fn get_auction(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<domain::settlement::Auction, error::Auction> {
+        let mut ex = self
+            .postgres
+            .pool
+            .begin()
+            .await
+            .context("begin")
+            .map_err(error::Auction::DbError)?;
+
+        let deadline = {
+            let scores = database::settlement_scores::fetch(&mut ex, auction_id)
+            .await
+            .context("fetch scores").map_err(error::Auction::DbError)?
+            // if score is missing, no competition / auction exist for this auction_id
+            .ok_or(error::Auction::Missing)?;
+
+            (scores.block_deadline as u64).into()
+        };
+
+        let prices = {
+            let db_prices = database::auction_prices::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch auction prices")
+                .map_err(error::Auction::DbError)?;
+
+            let mut prices = HashMap::new();
+            for price in db_prices {
+                let token = eth::H160(price.token.0).into();
+                let price = big_decimal_to_u256(&price.price)
+                    .ok_or(domain::auction::InvalidPrice)
+                    .and_then(|p| domain::auction::Price::new(p.into()))
+                    .map_err(error::Auction::Price)?;
+                prices.insert(token, price);
+            }
+            prices
+        };
+
+        let orders = {
+            // get all orders from a competition auction
+            let auction_orders = database::auction_orders::fetch(&mut ex, auction_id)
+                .await
+                .context("fetch auction orders")
+                .map_err(error::Auction::DbError)?
+                .ok_or(error::Auction::Missing)?
+                .into_iter()
+                .map(|order| domain::OrderUid(order.0))
+                .collect::<HashSet<_>>();
+
+            // get quotes for all auction orders
+            let quotes = self
+                .postgres
+                .read_quotes(auction_orders.iter())
+                .await
+                .map_err(error::Auction::DbError)?;
+
+            // get fee policies for all orders that were part of the competition auction
+            let policies = database::fee_policies::fetch_all_for_auction(&mut ex, auction_id)
+                .await
+                .context("fetch fee policies")
+                .map_err(error::Auction::DbError)?;
+
+            // compile order data
+            let mut orders = HashMap::new();
+            for order in auction_orders.iter() {
+                let quote = quotes
+                    .get(order)
+                    .ok_or(error::Auction::MissingQuote(*order))?;
```

Comment:
I think we still have orders without quotes in the DB. `MissingQuote` error should only be returned if there is no quote for the `PriceImprovement` fee policy.

Reply:
The orders are [not included](https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/solvable_orders.rs#L268) in the `Auction` if they don't have a `Quote`.

I believe this was intentional decision by the team.

But still fixed, as this part of the code might change.